Data/Sys/Shaders/PerceptualHDR.glsl

```diff
@@ -0,0 +1,72 @@
+/*
+[configuration]
+
+[OptionRangeFloat]
+GUIName = Desaturation
+OptionName = DESATURATION
+MinValue = 0.0
+MaxValue = 1.0
+StepAmount = 0.1
+DefaultValue = 0.0
+
+[/configuration]
+*/
+
+/***** Linear <--> Oklab *****/
+
+const mat4 RGBtoLMS = mat4(
+    0.4122214708, 0.2119034982, 0.0883024619, 0.0000000000,
+    0.5363325363, 0.6806995451, 0.2817188376, 0.0000000000,
+    0.0514459929, 0.1073969566, 0.6299787005, 0.0000000000,
+    0.0000000000, 0.0000000000, 0.0000000000, 1.0000000000);
+
+const mat4 LMStoOklab = mat4(
+    0.2104542553, 1.9779984951, 0.0259040371, 0.0000000000,
+    0.7936177850, -2.4285922050, 0.7827717662, 0.0000000000,
+    -0.0040720468, 0.4505937099, -0.8086757660, 0.0000000000,
+    0.0000000000, 0.0000000000, 0.0000000000, 1.0000000000);
+
+float4 LinearRGBToOklab(float4 c)
+{
+    return LMStoOklab * pow(RGBtoLMS * c, float4(1.0 / 3.0));
```

Comment:
Any reason you are also converting the alpha channel?

Reply:
It is unaffected throughout this process. It is reversed at the end.